src/components/portrait/portrait.component.tsx

```diff
@@ -82,9 +85,17 @@ const Portrait = ({
   ...rest
 }: PortraitProps) => {
   const [externalError, setExternalError] = useState(false);
+  const [isGravatarLoaded, setGravatarIsLoaded] = useState(false);
   const { roundedCornersOptOut } = useContext(RoundedCornersOptOutContext);
   const defaultShape = roundedCornersOptOut ? "square" : "circle";
 
+  if (isGravatarLoaded && !deprecatedGravatarWarnTriggered) {
+    deprecatedGravatarWarnTriggered = true;
+    Logger.deprecate(
+      "The `gravatar` prop has been deprecated and will soon be removed."
+    );
+  }
+
```

Comment:
**suggestion**: since [`Logger.deprecate` will be excluded from a consumer's production build](https://github.com/Sage/carbon/blob/master/src/__internal__/utils/logger/index.ts#L3), this new state won't control anything, so I'd suggest we replace the state with the prop to reduce the number of re-renders.

```suggestion
  if (gravatar && !deprecatedGravatarWarnTriggered) {
    deprecatedGravatarWarnTriggered = true;
    Logger.deprecate(
      "The `gravatar` prop has been deprecated and will soon be removed."
    );
  }

```

Reply:
The reason why I've introduced a new state variable is because we only want to fire the warning when the `gravatar` prop is used. 

For example, `Profile` renders `Portrait`, if you pass the `email` prop this is passed down to `Portrait` as the `gravatar` prop. Which means consumers may get warnings for the use of the `gravatar` prop even without using it intentionally. This way the warning will only fire if the `gravatar` prop is passed, and a gravatar image is actually loaded 👍